src/ui_menu_nbgl.c

```diff
@@ -0,0 +1,101 @@
+/*******************************************************************************
+ *   Ledger App - Bitcoin Wallet
+ *   (c) 2022 Ledger
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ********************************************************************************/
+
+#ifdef HAVE_NBGL
+#include "btchip_display_variables.h"
+#include "btchip_internal.h"
+#include "ui_nbgl.h"
+
+#include "btchip_bagl_extensions.h"
+#include "nbgl_use_case.h"
+
+#define NB_INFO_FIELDS 2
+#define PAGE_START 0
+#define NB_PAGE_SETTING 2
+#define IS_TOUCHABLE true
+
+#define NB_SETTINGS_SWITCHES 1
+#define SWITCH_PUBLIC_KEY_SET_TOKEN FIRST_USER_TOKEN
+
+static char text[20];
+static char title[20];
+static const char *const infoContents[] = {APPVERSION, "(c) 2022 Ledger"};
+
+static nbgl_layoutSwitch_t switches[NB_SETTINGS_SWITCHES];
+
+static void quit_cb(void) {
+    os_sched_exit(-1);
+}
+
+static void switch_public_key(void) {
+  uint8_t value = (N_btchip.pubKeyRequestRestriction ? 0 : 1);
```

Comment:
uint8_t value = (N_btchip.pubKeyRequestRestriction != 0) ? 0 : 1; would be more readable and pubKeyRequestRestriction is not really a boolean